apps/camera_pipe/CMakeLists.txt

```diff
@@ -12,15 +12,12 @@ set(CMAKE_CXX_EXTENSIONS NO)
 find_package(Halide REQUIRED)
 
 # Generator
-add_executable(camera_pipe.generator camera_pipe_generator.cpp)
-target_link_libraries(camera_pipe.generator
-                      PRIVATE
-                      Halide::Generator
-                      Halide::Tools)
+add_halide_generator(camera_pipe.generator SOURCES camera_pipe_generator.cpp)
+target_link_libraries(camera_pipe.generator PRIVATE Halide::Tools)
```

Comment:
```
if (TARGET camera_pipe.generator)
   target_link_libraries(camera_pipe.generator PRIVATE Halide::Tools)
endif ()
```
Though it wouldn't be unreasonable to link to `Halide::Tools` by default or to include an argument to `add_halide_generator` that calls `target_link_libraries` appropriately behind the scenes.

Reply:
Why do we need to check to see if it's a target... didn't we just add it as a target?

Reply:
also, this doesn't seem like the right fix -- if it somehow isn't a target, it won't build properly without this

Reply:
When cross-compiling, the fully qualified name will exist as an imported target. When doing a host build, the unqualified name will _also_ exist as a build target.